### ISSUE TITLE
New regex filter for Jinja2 expressions

### ISSUE THREAD

**Issue by Mierdin** [StackStorm/st2#3343]
In https://github.com/StackStorm/st2/blob/master/st2common/st2common/jinja/filters/regex.py, there are 3 possible filters that can be used in a Jinja2 expression in a rule. Unfortunately, the only function that returns a string is `regex_replace`. A user in the slack community wanted to be able to filter some text from a trigger through a filter, and use the resulting value - sort of like `regex_search` but returning the matched string, instead of a boolean.

So basically we need to add a filter that does this. The functionality of `regex_search` but returning the matched string instead of a boolean - and it would be nice if it would honor capture groups as well.

**Comment by armab** [comment]
:+1: for more useful filters. That's one of the advantages that makes Ansible so useful, [for example](http://docs.ansible.com/ansible/playbooks_filters.html).

> Also not similar, but related: https://github.com/StackStorm/st2/issues/2956 will be to allow users defining their cusom filters "on fly".
